---
title: "Coexpression plots for He scRNA-seq project"
output: html_document
date: "`r format(Sys.Date(), format ='%d %B %Y')`"
author: "Greg Wray"
---

### Introduction
Code for generating coexpression plots for *Heliocidaris erythrogramma* embryonic single-cell time course. Plots featured in Figure 7 and Supplementary Figure ## in Massri et al. manuscript.

### R Code

Takes as input 2 coexpression count tables generated by custom Python code.
Assumes input files are in the same R project.

```{r import packages, include=FALSE}
library(tidyverse)
```

```{r import data}
# read in data and convert to tibble
He_df <- read.csv("He_tc_ordered.csv") |> as_tibble()
Lv_df <- read.csv("Lv_tc_ordered.csv") |> as_tibble()
```

```{r reformat data}
# remove index, create species column, reorder columns
He_df <- select(He_df, -X)
He_df <- mutate(He_df, species = "He")
He_df <- select(He_df, species, everything())
Lv_df <- select(Lv_df, -X)
Lv_df <- mutate(Lv_df, species = "Lv")
He_df <- select(He_df, species, everything())
```

```{r merge tables}
# combine tables from 2 species and pivot to long
both_df <- rbind(Lv_df, He_df)
long_tc <- both_df |> 
  pivot_longer(cols = starts_with("X"), values_to = "coex")
glimpse(long_tc)
```

```{r add time values}
# define time points and add hpf column
time_pts <- c(6,9,12,16,20,24)
n_t_pts <- length(time_pts)
len_long_tc <- nrow(long_tc)
long_tc = mutate(long_tc, hpf = rep(time_pts, len_long_tc/n_t_pts))
long_tc <- select(long_tc, !name)
glimpse(long_tc)
```

```{r generate plot}
# instructions: provide gene names g1 and g2 in alphabetical order
# produces a line plot of specified gene pair for both species
g1 = 'alx1'
g2 = 'delta'
long_tc |>
  filter(gene_1 == get("g1") & gene_2 == get("g2")) |>
  ggplot(aes(hpf, coex, color=species)) +
        geom_line(size = 1) +
        labs(y = '% cells with coexpression') +
        labs(title = str_c(g1, ' and ', g2)) +
        theme_minimal()
```

```{r save plot, eval=FALSE}
# instructions: change the path and name of the comparison set as needed
# writes the plot to the "Plots" folder
spath = './Comparison_sets/SkC/Plots'
sname = str_c(g1, '_', g2, '.pdf')
#ggsave(sname, path = spath)
#ggsave(sname, path = spath, height = 1, width = 1, units = 'cm')
ggsave(sname, path = spath, units = 'cm')
```

